app/migrations/Version20180502142950.php

```diff
@@ -0,0 +1,85 @@
+use Doctrine\DBAL\Migrations\AbstractMigration;
+use Doctrine\DBAL\Schema\Schema;
+use Ramsey\Uuid\Uuid;
+use Symfony\Component\DependencyInjection\ContainerAwareInterface;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+class Version20180502142950 extends AbstractMigration implements ContainerAwareInterface
+{
+    /**
+     * @var DonationRequestUtils
+     */
+    private $donationRequestUtils;
+
+    public function up(Schema $schema): void
+    {
+        $this->addSql('CREATE INDEX donation_uuid_idx ON donations (uuid)');
+        $this->addSql('ALTER TABLE donations ADD status VARCHAR(25) DEFAULT NULL, CHANGE donated_at updated_at DATETIME DEFAULT NULL, ADD paybox_order_ref VARCHAR(255) DEFAULT NULL');
+        $results = $this->connection->executeQuery('SELECT uuid, CONCAT(first_name, \' \', last_name) AS full_name FROM donations')->fetchAll();
+
+        foreach ($results as $result) {
+            ['uuid' => $uuid, 'full_name' => $fullName] = $result;
+            $slug = $this->donationRequestUtils->buildDonationReference(Uuid::fromString($uuid), $fullName);
+            $this->addSql("UPDATE donations SET paybox_order_ref = '$slug' WHERE uuid = '$uuid'");
+        }
+        $this->addSql('UPDATE donations SET donations.status = \'finished\' WHERE donations.finished = 1');
+        $this->addSql('UPDATE donations SET donations.status = \'waiting_confirmation\' WHERE donations.finished = 0');
+        $this->addSql('UPDATE donations SET donations.status = \'subscription_in_progress\' WHERE donations.duration != 0');
+        $this->addSql('UPDATE donations SET donations.updated_at = donations.created_at WHERE donations.updated_at IS NULL');
+
+        $this->addSql('CREATE TABLE transaction (id INT UNSIGNED AUTO_INCREMENT NOT NULL, donation_id INT UNSIGNED DEFAULT NULL, paybox_result_code VARCHAR(100) DEFAULT NULL, paybox_authorization_code VARCHAR(100) DEFAULT NULL, paybox_payload JSON DEFAULT NULL COMMENT \'(DC2Type:json_array)\', paybox_date_time DATETIME DEFAULT NULL, paybox_transaction_id VARCHAR(255) DEFAULT NULL, paybox_subscription_id VARCHAR(255) DEFAULT NULL, created_at DATETIME NOT NULL, UNIQUE INDEX UNIQ_723705D15A4036C7 (paybox_transaction_id), INDEX IDX_723705D14DC1279C (donation_id), PRIMARY KEY(id)) DEFAULT CHARACTER SET utf8 COLLATE utf8_unicode_ci ENGINE = InnoDB');
+        $this->addSql('ALTER TABLE transaction ADD CONSTRAINT FK_723705D14DC1279C FOREIGN KEY (donation_id) REFERENCES donations (id)');
+        $this->addSql(
+            'INSERT INTO transaction (donation_id, paybox_result_code, paybox_authorization_code, paybox_payload, paybox_date_time, paybox_transaction_id, created_at)
+                    (SELECT id, paybox_result_code, paybox_authorization_code, paybox_payload,
+                      IF (paybox_payload->"$.date" IS NOT NULL AND paybox_payload->"$.time" IS NOT NULL, STR_TO_DATE(
+                        CONCAT(
+                          SUBSTR(paybox_payload->"$.date", 2, 2),
```

Comment:
Please use second arg to pass parameters to the query

Reply:
It's not a parameter ! It's mysql way to read and use json data